### DISCUSSION TITLE
Migrate data from AWS S3 to own min.io server

### DISCUSSION THREAD

**Discussion by nguyendatdh93** [minio/minio#18004] [Q&A] [answered]
Hi all!

I want to move all objects in a bucket fro aws s3 to min.io server. I used the commands `mc mirror` and `mc cp -r`. it works well. I also check the folders on server. They have all files like aws s3. But when i access to min.io console. It just show folders. Files don't show in folder like s3. 

Does anyone know this issue? 

Thanks

**Comment by nguyendatdh93** [comment]
btw, when i upload a file directly from min.io console. File saved on server like this structure. 
![image](https://github.com/minio/minio/assets/22823500/82aad3fb-8310-430e-ba4d-92eb44af82fe)
But if i used command `mc cp -r` or `mc mirror`. It likes original file. 
![image](https://github.com/minio/minio/assets/22823500/b433f4e5-91ba-4771-af0a-edd3d2a8892a)

**Comment by harshavardhana** [comment, marked as answer]
Use batch replication
